### ISSUE TITLE
Error in post-command-hook (ivy--queue-exhibit): (end-of-buffer)

### ISSUE THREAD

**Issue by nivaca** [abo-abo/swiper#2692]
I get the error

    Error in post-command-hook (ivy--queue-exhibit): (end-of-buffer)

when I run swiper twice in order to repeat the previous search (`S-s, S-s` with my key bindings)
Have I configured something wrong, or is it that to repeat the previous search I have to do otherwise?
Thanks!

My config is available here: https://pastebin.com/nts6Nz9L

**Comment by nivaca** [comment]
Thanks for your message. How can I use Ivy when running `emacs -Q`? No packages get loaded in such a way.

**Comment by basil-conto** [comment]
> How can I use Ivy when running `emacs -Q`? No packages get loaded in such a way.

<kbd>M-x</kbd>`package-initialize`<kbd>RET</kbd>

After this, you can either call autoloaded functions directly, or explicitly load packages via e.g. <kbd>M-x</kbd>`load-library`<kbd>RET</kbd>`swiper`<kbd>RET</kbd>.

**Comment by nivaca** [comment]
Hi, thanks for your message.
I already tried that, but neither Swiper nor its functions are available to me in that way. Sorry.

**Comment by basil-conto** [comment]
How have you installed Swiper?

If the default installation doesn't work for you, you can try downloading the sources from this repository and running `make plain` from inside the checkout.

**Comment by nivaca** [comment]
Yes, it's installed and I use it everyday. But do you mean I have to install it again when running `emacs -Q`?

**Comment by basil-conto** [comment]
I asked *how* it's installed, not *whether*.

For example, if you've installed the `swiper` package in the default `package-user-dir` via the default built-in `package.el` manager (which based on your posted config I'm assuming you have), then `emacs -Q` followed by <kbd>M-x</kbd>`package-initialize`<kbd>RET</kbd> should allow you to call the autoloaded command <kbd>M-x</kbd>`swiper`<kbd>RET</kbd>.  If it doesn't, then something's wrong or unusual in your setup. Perhaps uninstalling and reinstalling the `swiper` package will help in that case.

In any case, it's not crucial that you follow these exact steps---they're just usually the simplest---as long as there's a minimal reproducible recipe, ideally without having to evaluate your entire config.

**Comment by nivaca** [comment]
I install all my packages with `package.el`, as you noticed. The problem was that the packages aren't installed in the default directory. I have correctly set the variable, and the packages were correctly initialized.

Now, after Emacs starts, I run these commands in the scratch buffer:
```
(setq package-user-dir "~/emacs/packages/pc")
(global-set-key (quote [f2]) 'counsel-grep-or-swiper)
```

Then I press "F2" twice, and I get the following message:
`ivy-read: Command attempted to use minibuffer while in minibuffer`

whereas, using my config, I get the said error. There must be something in my config concerning the minibuffer config, but I haven't found it yet.

**Comment by basil-conto** [comment]
> Then I press "F2" twice, and I get the following message:
`ivy-read: Command attempted to use minibuffer while in minibuffer`

That is because Emacs does not allow recursive minibuffers by default. If you want to call `counsel-grep-or-swiper` twice in succession, i.e. once from some buffer and then a second time from within the minibuffer, then you need:

```el
(setq enable-recursive-minibuffers t)
```

> whereas, using my config, I get the said error.

If I enable recursive minibuffers as shown above, and press <kbd>F2</kbd> twice, I get the expected nested Swiper completion session, not an error in `post-command-hook`.

> There must be something in my config concerning the minibuffer config, but I haven't found it yet.

It would be most useful if you could pinpoint that. Thanks.

**Comment by nivaca** [comment]
Yes, thanks for your help. If I discover it, I'll reopen the issue.

**Comment by mekeor** [comment]
i am also experiencing this issue (since months) but i suspect it's not a bug but just a change: formerly, it was possible to resume a search by typing `C-s C-s` but now it doesn't. if you have `(setq enable-recursive-minibuffers t)`, then it will launch a search within the mini-buffer. otherwise it should lead to the message `ivy-read: Command attempted to use minibuffer while in minibuffer`. it does show this message for me when i use `emacs -Q`. but instead of that message, i get `Error in post-command-hook (ivy--queue-exhibit): (end-of-buffer)`.

anyway, my problem is not that i get an error instead of a message. i'd just love to resume my search with `C-s C-s`, as i could do formerly. how do i achieve that?

(this is how i installed swiper:)

```emacs-lisp

(setq straight-use-package-by-default t)


;; straight package-manager: packages as git-repositories.

(defvar bootstrap-version)
(let ((bootstrap-file
          (expand-file-name "straight/repos/straight.el/bootstrap.el" user-emacs-directory))
         (bootstrap-version 5))
    (unless (file-exists-p bootstrap-file)
        (with-current-buffer
            (url-retrieve-synchronously
                "https://raw.githubusercontent.com/raxod502/straight.el/develop/install.el"
                'silent 'inhibit-cookies)
            (goto-char (point-max))
            (eval-print-last-sexp)))
    (load bootstrap-file nil 'nomessage))


;; https://github.com/raxod502/straight.el/issues/548#issuecomment-663135875
(straight--package-built-in-p 'nadvice)
(puthash 'nadvice t straight--cached-built-in-packages)

;; use-package

(straight-use-package 'use-package)

;; Ivy (smart choice tool)

(use-package ivy
    :delight ivy-mode
    :config
    (ivy-mode 1)
    (setq ivy-use-virtual-buffers t)
    (setq enable-recursive-minibuffers nil))

;; Counsel (use ivy everywhere)

(use-package counsel
    :bind*
    ("C-d p s" . counsel-git-grep)
    :delight counsel-mode
    :config
    (counsel-mode 1)
    (setq counsel-find-file-at-point t))

;; Swiper (use ivy for search)

(use-package swiper
    :bind*
    ("C-s" . swiper)
    ("C-r" . swiper))
```

emacs version: `GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.23, cairo version 1.16.0)
 of 2020-11-08, modified by Debian`

**Comment by mekeor** [comment]
@nivaca, could you re-open the issue maybe? :)

**Comment by nivaca** [comment]
@mekeor I have reopened the issue.

**Comment by basil-conto** [comment]
@mekeor Sorry, it's not immediately clear to me what you mean.  Could you please open a separate bug report with a step-by-step recipe for reproducing the issue? Thanks.

**Comment by basil-conto** [comment]
> @mekeor Sorry, it's not immediately clear to me what you mean.

I cannot reproduce the error based only on your description, so..

> Could you please open a separate bug report with a step-by-step recipe for reproducing the issue? Thanks.

...in the absence of further information, I'm once again closing this issue.
